app/main.py

```diff
@@ -1,2 +1,119 @@
+from typing import Any, Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any,
+                 next_node: "Node" = None) -> None:
+        self.key = key
+        self.value = value
+        self.next_node = next_node
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8,
+                 load_factor: float = 0.66) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._hash(key)
+        if self.table[index] is None:
+            self.table[index] = Node(key, value)
+
+        else:
+            current = self.table[index]
+            while current is not None:
+                if current.key == key:
+                    current.value = value
+                    return
+                if current.next_node is None:
+                    break
+                current = current.next_node
+            current.next_node = Node(key, value)
+        self.size += 1
+
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._hash(key)
+        current = self.table[index]
+        while current is not None:
+            if current.key == key:
```

Comment:
Compare hash of key too

Reply:
Done:
if current.key == key and hash(current.key) == hash(key):